src/_11ty/utils/get-page-info.js

```diff
@@ -0,0 +1,16 @@
+export function getPageInfo(pages, url) {
+  const cached = _cachedPages.get(url);
+  if (cached) {
+    return cached;
+  }
+
+  for (const page of pages) {
+    if (page.url === url) {
+      _cachedPages.set(url, page);
+      return page;
+    }
+  }
+}
+
+// TODO(parlough): Replace this functionality or document it.
+const _cachedPages = new Map();
```

Comment:
would love to learn how this cache works!

Reply:
Me too. I'm hoping I can make it unnecessary, but if not, I'll document it clearly!!